### DISCUSSION TITLE
How to delete all previous messages in chat with the user?

### DISCUSSION THREAD

**Discussion by TimurNurlygayanov** [python-telegram-bot/python-telegram-bot#2273] [Q&A]
Desired behavior:
When a user uses /start, the bot should delete all previous messages in the chat and start the new conversation

But it is not clear how to do this with python-telegram-bot library

**Comment by Bibo-Joshi** [comment]
See [here](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-Asked-Questions#can-you-add-feature-to-ptb-can-i-do-thing-with-my-bot), i.e. there is no way to clear a complete chat via the bot API. Even if you kept track of all message IDs sent in that chat manually, you can only detele messages up to [48hs](https://core.telegram.org/bots/api#deletemessage).

**Reply by TimurNurlygayanov** [reply]
Thank you )
